web/src/pages/login.tsx

```diff
@@ -0,0 +1,72 @@
+import Button from "../components/Inputs/Button";
+import TextField from "../components/Inputs/TextField";
+import { useState, ChangeEvent, useEffect } from 'react';
+
+import { MdArrowBackIosNew } from "react-icons/md";
+
+import IconButton from "../components/Inputs/IconButton";
+import LinkIconButton from "../components/Inputs/LinkIconButton";
+import Link from "next/link";
+
+
+export default function LoginPage() {
+
+    const [username, setUsername] = useState("");
+    const [password, setPassword] = useState("");
+    
+    const handleUsernameChange = (e: ChangeEvent<HTMLInputElement>) => {
+        setUsername(e.target.value);
+    }
+
+    const handlePasswordChange = (e: ChangeEvent<HTMLInputElement>) => {
+        setPassword(e.target.value);
+    }
+
+    useEffect(() => {
+        setPassword("");
+        setUsername("");
+    }, [])
+
+    return (
+        <div className={"h-screen"}>
+            <div className={"mb-12"}>
+                <nav className={"sticky top-0 mb-16 h-12 rounded-b-xl px-4"}>
+                    <ul className={"z-10 flex h-full list-none justify-self-start"}>
+                        <LinkIconButton href={"/index"}>
+                            <MdArrowBackIosNew className={"h-full w-8"}/>
+                        </LinkIconButton>
+                    </ul>
+                </nav>
+            </div>
+            
+            <main className={"container mx-auto w-5/12"}> 
+                <p className="text-4xl text-center">Log in</p><br/>
+                <div className={"m-10 flex-col items-center justify-center"}>
+                    
+                    <form> 
+                        <label>
+                            <TextField label="Enter username or email:" placeholder="Username or Email" onChange={handleUsernameChange} value={username}></TextField> 
+                        </label><br/>
+                        <label> 
+                                {/*@ts-expect-error FIXME typing bug*/} 
+                                <TextField label="Enter password:" type="password" placeholder="Password" onChange={handlePasswordChange} value={password}></TextField> 
+                        </label><br/><br/>
+                        <div className="flex justify-center"> 
+                            <Link href="/feed">
+                                <Button onClick={() => { console.log("Logged in as: " + {username} + " with password: " + {password})}}>Continue</Button>
```

Comment:
When using Link it is recommended by Next that you have an <a></a> component inside instead of a button.

In this case I would recommend you remove the Link altogether and keep the button but instead of console logging that onClick, you can pass in a handle function that will use the router to redirect the user to the feed page:


```
function Example() {
    const router = useRouter

    const handleContinueClick = () {
        router.replace('/feed')
    }

    return <Button onClick={handleContinueClick}>Continue</Button>
}
```